File: nahyun/Re1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int count1 = 0;
        int count2 = 0;
        
        boolean check = true;
        for(int i = 0; i<goal.length; i++) {
            if(!check) break;
            //cards1에서 꺼낼 수 있을 때
            if(count1 < cards1.length && goal[i].equals(cards1[count1])) {
                count1++;
                continue;
            }
            //cards2에서 꺼낼 수 있을 때
            else if(count2 < cards2.length && goal[i].equals(cards2[count2])) { 
                count2++;
                continue;
            }
            check = false;
            break;
        }
        
        if(check) return "Yes";
        else return "No";
    }
}

File: minseo/Re1/카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
  let goal1 = goal.filter(word => cards1.includes(word));
  let goal2 = goal.filter(word => cards2.includes(word));

 
  let fix1 = cards1.slice(0, goal1.length);
  let fix2 = cards2.slice(0, goal2.length);

  if (goal1.join(" ") !== fix1.join(" ") || goal2.join(" ") !== fix2.join(" ")) {
    return "No";
  }
  return "Yes";
}
